src/utils/resources/vm/hooks/disk/useDisksTableData.ts

```diff
@@ -59,10 +58,9 @@ const useDisksTableData: UseDisksTableDisks = (vm, vmi) => {
     const isInstanceTypeVM = Boolean(getInstanceTypeMatcher(vm));
 
     const diskDevices: DiskRawData[] = (vmVolumes || []).map((volume) => {
-      const disk =
-        isInstanceTypeVM && Boolean(volume.name === ROOTDISK)
-          ? { name: ROOTDISK }
-          : vmDisks?.find(({ name }) => name === volume?.name);
+      let disk = vmDisks?.find(({ name }) => name === volume?.name);
+
+      if (!disk && isInstanceTypeVM) disk = { name: volume?.name };
```

Comment:
What happens if !disk and !IT ?

Reply:
in the vm instancetype some disks are implicit like rootdisk and cloudinit. The vm has the volume but not the disk specified. 
So i created this condition where if the disk is not found, lets just assume that is implictly declared with default values.